frame/support/src/traits/filter.rs

```diff
@@ -183,7 +183,7 @@ macro_rules! impl_filter_stack {
 /// Type that provide some integrity tests.
 ///
 /// This implemented for modules by `decl_module`.
-#[impl_trait_for_tuples::impl_for_tuples(30)]
+#[impl_trait_for_tuples::impl_for_tuples(100)]
```

Comment:
Can you do some simple comparison of `cargo build -p frame-support` between 30 and 100 as parameters here.

Reply:
master: `Finished dev [unoptimized + debuginfo] target(s) in 37.59s`
this branch: ` Finished dev [unoptimized + debuginfo] target(s) in 46.04s`

both after a cargo-clean, done in my humble macbook one after another.

Reply:
Does this make an implicit limit of number of pallets a runtime can have? Should we just make it 256 if it doesn't completely destroy the compiler?

Reply:
Yes it does this, so please Ser @kianenigma, try with 256.

Reply:
There are more instances of `#[impl_trait_for_tuples::impl_for_tuples(30)]` in the code base.  
Do they also need to be updated? Then we should maybe add a constant for it like `MAX_NUM_PALLETS`.

Reply:
Moved all of the ones in frame-support (previous number only moved like half of them to 100) and now I get: 

`Finished dev [unoptimized + debuginfo] target(s) in 2m 54s`

Seems a bit too much. Even worse, you need frame-support for all pallets so building a runtime is now also a lot slower. 

So even when building the entire runtime, we feel the overhead

Nonetheless, I see only two options: 

- We move to flat pallets, and only implement up to 128 or some smaller value to not cause pain for everyone. 
- We can feature gate frame-support to generate these based on the bound needed


If we don't merge this PR, we should actually reduce all of them to 2, or maximum 8 for manual cases, because with nested tuples, you only ever need `impl Trait for (T)` and `impl trait for (T, T)`. All the rest is redundant.

Reply:
making the default be 64, and having `tuples-96`, `tuples-128`, `tuples-160` features seems like a reasonable option.

Reply:
applied my suggestion, not too bad, and the compile time (with the default 64) is ` 53.31`. This will further come lower once we do https://github.com/paritytech/substrate/issues/11925.


> then we should maybe add a constant for it like MAX_NUM_PALLETS.

constants are not available at macro expansion phase afaik.